bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/AbstractMQTTThingHandler.java

```diff
@@ -61,13 +75,17 @@
  * @author David Graeff - Initial contribution
  */
 @NonNullByDefault
-public abstract class AbstractMQTTThingHandler extends BaseThingHandler implements ChannelStateUpdateListener {
+public abstract class AbstractMQTTThingHandler extends BaseThingHandler
+        implements ChannelStateUpdateListener, AvailabilityTracker {
     private final Logger logger = LoggerFactory.getLogger(AbstractMQTTThingHandler.class);
     // Timeout for the entire tree parsing and subscription
     private final int subscribeTimeout;
 
     protected @Nullable MqttBrokerConnection connection;
 
+    private AtomicBoolean messageReceived = new AtomicBoolean(false);
+    private Map<String, @Nullable ChannelState> availabilityStates = new ConcurrentHashMap<>();
```

Comment:
Why "@Nullable"? Wouldn't it be more easy to make this non nullable?

Reply:
I have 2 problems with `Map` if the value is `@NonNull`:
- `Map.get()` is assumed to be `@NonNull``, which is wrong. So I get warnings when doing a null check. And I do not like doing a `contains()` first, because that would double the compution time.
- `Map.computeIfPresent()` will not allow we to return `null` to remove the element. But this is the best way to do it in an atomic matter. (`ConcurrentHashMap` does not allow `null` as key or value)